### ISSUE TITLE
Using sdk with Python gives an SDKError

### ISSUE THREAD

**Issue by sauravganguli** [looker-open-source/sdk-codegen#521]
I am using sdk version 7.20.0 with Python 3.9.2 but all of a sudden it has stopped working for me.
I try to run the below code:
```
import looker_sdk
sdk = looker_sdk.init31("looker.ini")
sdk.me()
```

In the past this used to work for me but since yesterday its giving me an error. See below the error traceback. I am guessing this has something to do with my laptop settings that is not allowing me to authenticate the client. But I am unable to get to the root cause of the error. As a fallback, I did a clean reinstallation of Python and all other related packages, but that still doesn't solve the issue. The .ini file is also setup correctly and the url endpoint seems fine because same setup if running fine on my colleague's laptop. 
I am clueless for the last two days about what could be causing this issue suddenly. I know its difficult to get a conclusive solution this way, but any help would be highly appreciated.

> ---------------------------------------------------------------------------
> SDKError                                  Traceback (most recent call last)
> <ipython-input-4-5eaf8494302d> in <module>
>       3 sdk = looker_sdk.init31("looker.ini")
>       4 
> ----> 5 sdk.me()
> 
> ~/opt/anaconda3/lib/python3.7/site-packages/looker_sdk/sdk/api31/methods.py in me(self, fields, transport_options)
>    8088             models.User,
>    8089             query_params={"fields": fields},
> -> 8090             transport_options=transport_options,
>    8091         )
>    8092         assert isinstance(response, models.User)
> 
> ~/opt/anaconda3/lib/python3.7/site-packages/looker_sdk/rtl/api_methods.py in get(self, path, structure, query_params, transport_options)
>     148             body=None,
>     149             authenticator=self.auth.authenticate,
> --> 150             transport_options=transport_options,
>     151         )
>     152         return self._return(response, structure)
> 
> ~/opt/anaconda3/lib/python3.7/site-packages/looker_sdk/rtl/requests_transport.py in request(self, method, path, query_params, body, authenticator, transport_options)
>      65         timeout = self.settings.timeout
>      66         if authenticator:
> ---> 67             headers.update(authenticator())
>      68         if transport_options:
>      69             if transport_options.get("headers"):
> 
> ~/opt/anaconda3/lib/python3.7/site-packages/looker_sdk/rtl/auth_session.py in authenticate(self)
>      93             token = self._get_sudo_token()
>      94         else:
> ---> 95             token = self._get_token()
>      96 
>      97         return {"Authorization": f"Bearer {token.access_token}"}
> 
> ~/opt/anaconda3/lib/python3.7/site-packages/looker_sdk/rtl/auth_session.py in _get_token(self)
>      82         """Returns an active token."""
>      83         if not self.is_authenticated:
> ---> 84             self._login()
>      85         return self.token
>      86 
> 
> ~/opt/anaconda3/lib/python3.7/site-packages/looker_sdk/rtl/auth_session.py in _login(self)
>     140                 body=serialized,
>     141                 transport_options={
> --> 142                     "headers": {"Content-Type": "application/x-www-form-urlencoded"}
>     143                 },
>     144             )
> 
> ~/opt/anaconda3/lib/python3.7/site-packages/looker_sdk/rtl/auth_session.py in _ok(self, response)
>     208     def _ok(self, response: transport.Response) -> str:
>     209         if not response.ok:
> --> 210             raise error.SDKError(response.value)
>     211         return response.value.decode(encoding="utf-8")
>     212 
> 
> SDKError: b'{"message":"Not found","documentation_url":"http://docs.looker.com/"}'

**Comment by joeldodge79** [comment]
yeah, our API is pretty opaque around authentication errors (security by obscurity?) - I'm pretty sure this means the API3 credentials you're using (`CLIENT_ID` and `CLIENT_SECRET`) are no longer valid (or the user is disabled) on the server your hitting

**Comment by sauravganguli** [comment]
Hi @joeldodge79 it doesn't seem to be a credential issue. The same credentials work with the lookerapi, but only shows error when using the looker SDK. Also to verify I used another looker ini file from my colleague which works perfectly fine on her machine. But it shows the same error when I run that on my machine. Is there a way to send the credentials as parameters in the sdk init call, other than using ini file or env vars?

**Comment by joeldodge79** [comment]
I think we also support a `.netrc` file so maybe check if you have one of those (esp if you were playing with the ruby sdk which I think requires it)

if you have access see if you can add `print()` lines [here](https://github.com/looker-open-source/sdk-codegen/blob/6734edb21496bc44af68a3c26289cccf7c4d6095/python/looker_sdk/rtl/auth_session.py#L128) for the credentials as well as for the [f string here](https://github.com/looker-open-source/sdk-codegen/blob/6734edb21496bc44af68a3c26289cccf7c4d6095/python/looker_sdk/rtl/auth_session.py#L139) so you can actually see what credentials the code is sending and where it's sending them. You might check the [logs](https://docs.looker.com/admin-options/server/log) too. beyond that your best bet is opening a ticket with looker support

**Comment by sauravganguli** [comment]
Thanks for the mentioned steps, I am using the Python SDK only so the .netrc might not be applicable. I also checked first with Looker support and they too couldn't see any issues with the credentials. Sudoing my user, the script work fine in their machine. So we came to the conclusion that it is some machine setting which would be obstructing the init call, but nothing conclusive to know anything about the issue
